paddle/fluid/framework/details/multi_devices_graph_pass.cc

```diff
@@ -33,6 +33,92 @@
 namespace paddle {
 namespace framework {
 namespace details {
+namespace {
+void PolishGraphToSupportDataHazards(ir::Graph *graph) {
```

Comment:
I think `PolishGraphToSupportDataHazards` will be used in the `pass` of removing or adding nodes.